rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeStorageProviderImpl.java

```diff
@@ -91,4 +100,26 @@ public Optional<Coin> getLockingCap(ForBlock activations) {
             ))
         );
     }
+
+    @Override
+    public void setWeiTransferredToUnionBridge(co.rsk.core.Coin weiTransferred) {
+        if (weiTransferred != null && weiTransferred.compareTo(co.rsk.core.Coin.ZERO) < 0) {
+            throw new IllegalArgumentException("Wei transferred to Union Bridge cannot be negative");
+        }
+        this.weiTransferredToUnionBridge = weiTransferred;
+    }
+
+    @Override
+    public Optional<co.rsk.core.Coin> getWeiTransferredToUnionBridge(ForBlock activations) {
+        if (!activations.isActive(ConsensusRule.RSKIP502)) {
+            return Optional.empty();
```

Comment:
Probably no big deal but, should this throw an exception instead? This should never be attempted to be called prior to RSKIP502.

Reply:
This comment is also related to Marcos' comment. Let's discuss this further to find an appropriate solution.